internal/service/advancedclustertpf/resource_compatiblity.go

```diff
@@ -37,13 +39,21 @@ func findNumShardsUpdates(ctx context.Context, state, plan *TFModel, diags *diag
 	return planCounts
 }
 
-func resolveAPIInfo(ctx context.Context, plan *TFModel, diags *diag.Diagnostics, clusterLatest *admin.ClusterDescription20240805, client *config.MongoDBClient) *ExtraAPIInfo {
+func resolveAPIInfo(ctx context.Context, diags *diag.Diagnostics, client *config.MongoDBClient, plan *TFModel, clusterLatest *admin.ClusterDescription20240805, overrideUsingLegacySchema bool) *ExtraAPIInfo {
```

Comment:
[nit] `forceLegacySchema` less verbose?

Reply:
changed in: https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2896/commits/d39cf2f073eec031e73dc8f9b124673b5aca6855